src/lsm/manifest_level_fuzz.zig

```diff
@@ -0,0 +1,426 @@
+                var key = env.random.uintAtMostBiased(Key, table_count_max * 64);
+
+                while (insert_count < insert_max) : (insert_count += 1) {
+                    const table = env.generate_non_overlapping_table(key);
+                    try env.buffer.append(table);
+                    key = table.key_max;
+                }
+            }
+
+            const tables = env.buffer.items;
+            defer env.buffer.clearRetainingCapacity();
+
+            // Insert the generated tables into the ManifestLevel:
+            for (tables) |*table| {
+                assert(table.visible(env.snapshot));
+                assert(table.visible(lsm.snapshot_latest));
+                env.level.insert_table(&env.pool, table);
+            }
+
+            // Insert the generated tables into the Environment for reference:
+            for (tables) |*table| {
+                const index = binary_search.binary_search_values_raw(
+                    Key,
+                    TableInfo,
+                    key_min_from_table,
+                    compare_keys,
+                    env.tables.items,
+                    table.key_max,
+                    .{},
+                );
+
+                // Can't be equal as the tables may not overlap
+                if (index < env.tables.items.len) {
+                    assert(env.tables.items[index].key_min > table.key_max);
+                }
+
+                try env.tables.insert(index, table.*);
+            }
+        }
+
+        fn generate_non_overlapping_table(env: *Environment, key: Key) TableInfo {
+            var new_key_min = key + env.random.uintLessThanBiased(Key, 31) + 1;
+            assert(compare_keys(new_key_min, key) == .gt);
+
+            var i = binary_search.binary_search_values_raw(
+                Key,
+                TableInfo,
+                key_min_from_table,
+                compare_keys,
+                env.tables.items,
+                new_key_min,
+                .{},
+            );
+
+            if (i > 0) {
+                if (compare_keys(new_key_min, env.tables.items[i - 1].key_max) != .gt) {
+                    new_key_min = env.tables.items[i - 1].key_max + 1;
+                }
+            }
+
+            const next_key_min = for (env.tables.items[i..]) |table| {
+                switch (compare_keys(new_key_min, table.key_min)) {
+                    .lt => break table.key_min,
+                    .eq => new_key_min = table.key_max + 1,
+                    .gt => unreachable,
+                }
+            } else std.math.maxInt(Key);
+
+            const max_delta = @minimum(32, next_key_min - 1 - new_key_min);
+            const new_key_max = new_key_min + env.random.uintAtMostBiased(Key, max_delta);
+
+            return .{
+                .checksum = env.random.int(u128),
+                // zero addresses are used to indicate the table being removed.
```

Comment:
```suggestion
                // Zero addresses are used to indicate the table being removed.
```

Proper capitalization tells the reader we're sharp about the little things.